### ISSUE TITLE
Import CSV not working

### ISSUE THREAD

**Issue by XisoP01** [linuxserver/budge#4]
The .csv I exported from my bank can't be imported. After creating a bank account and clicking the import button I can select a .csv file I generated from my bank but after clicking Open in the selection pop-up I'm returned to the import landing wich is shown after clicking the import button..

Using a Dutch bank account. Maybe the issue occurs because the currency difference? Allready upvoted the adding of currency selection :)

**Comment by dugite-code** [comment]
@alex-phillips The CSV importer also appears to only support American style date formats `MM/DD/YYYY` rather than the common international `DD/MM/YYYY` 

Additionally the importer appears to require a payee, for example the headers I have in my banks csv is: `Date,Description,Debit/Credit,Balance` where the balance is often blank.

**Comment by Hukuma1** [comment]
I got similar as above (`Date,Description,Debit/Credit,Balance`) with the `Description` being the `Payee`. I also had to delete a few lines as my bank added some summary stuff to the CSV file at the very top. The very first line of your CSV file should be the headers only.

**Comment by dugite-code** [comment]
I would have thought the description is for the memo field rather than the payee, so maybe that's my misunderstanding. My old bank did the same as yours with adding extra summary lines (and a bunch of blank lines) at the top and bottom of the CSV file

**Comment by Hukuma1** [comment]
Yeah, unfortunately my bank is in bizarro world and just combines both into one field it seems. The payee and the memo is all in one field under description.

**Comment by adriancossu** [comment]
I'm also unable to get the CSV to upload. My headers are (`Date, Description, Debit, Credit, Balance`). My date is in the american format with month first. I've tried to the following mapping combinations

-  `Description` to both `payee`  and `memo`. 
-  `debit`  to `outflow` and to `amount`
-  `credit` to both `inflow` and `amount`

Unfortunately, none of these combinations have worked for me. Appreciate any advice. Thanks,

**Comment by willwillis** [comment]
Does someone have an example of what an "good" importable file looks like? It might be good to include a template or reference in the project.

**Comment by typhooncarr** [comment]
Not to make things more complicated however my bank is putting the date format as YYYYMMDD which breaks the import parser. This almost needs to be a user definable option as to what date format the CSV file has, which is how FireFly-iii Data Importer is working.

**Comment by Hukuma1** [comment]
I think it'd be smart to branch out each institution to its own template? I think FireFly does this too. e.g. Chase Bank import, Bank of America import, etc. Different countries too perhaps.

**Comment by DastardlyDino** [comment]
+1 Still not able to import CSV files.

Not being able to import CSV is a big issue for me personally. While ideally I would type in each transaction myself individually I am too lazy and like to import weekly or monthly CSV files.

**Comment by dugite-code** [comment]
@DastardlyDino The best option of ensuring you can import your transactions currently is to use the export functionality to generate a valid csv file to use as a template. By opening this csv file and your bank's csv file in a plain text editor you can then see what you need to do to modify your own transaction data.
